### ISSUE TITLE
proposed convention for CRL codes

### ISSUE THREAD

**Issue by rwmichaels** [JeffersonLab/analyzer#73]
CRL = CODA Readout Language, the software that reads VME and Fastbus.  

For writers of CRL for VME modules, I propose that we have a simple convention:  

The beginning of the readout of a Module has a Header, which is a unique 32-bit word that appears in the db_cratemap.dat.  This is already the case.  

Right after the Header, I propose to put the number of words for that module, which is always known by the CRL, but may not be easily knowable for Podd, depending on the Module.  Some modules have a fixed number of workds, but e.g. the FADC250 puts out a variable number of words, and it is less strain on the intelligence of the decoder to use this convention.  

To summarize, I propose a "Header" followed by "Num words".

**Comment by rwmichaels** [comment]
I'd like to amend this proposed convention.  It's fine if we use Headers, but I think it's better to always use Bank instead, going forward.  A "Bank" is a well defined concept in CODA.  Specifically, all the data from one Module (or one bank of identical Modules, like the FADC250) should go into one Bank, and no other data should go into the same Bank.  You might argue that other data could go into it if the decoding looks for the Block Header, Event Header, and Block Trailer and take everything in between.  These Headers are also well-defined in CODA.  However, extraneous words could get confused for a Header.  So put all other words into some other bank, please.  With this convention, we'll have no room for confusion in the decoding of the pipelining modules.
